site/source/docs/api_reference/val.h.rst

```diff
@@ -33,6 +33,19 @@ Guide material for this class can be found in :ref:`embind-val-guide`.
     val xhr = val::global("XMLHttpRequest").new_();
     xhr.call("open", std::string("GET"), std::string("http://url"));
 
+  You can test whether the ``open`` method call was successful using :cpp:func:`~emscripten::val::operator[]` to read an object property, then :cpp:func:`~emscripten::val::as` to coerce the type:
+
+  .. code:: cpp
+
+    const char* state;
+    switch (xhr["readyState"].as<int>()) {
+	case 0:
+      state = "UNSENT"; break;
```

Comment:
indentation looks a little odd here (the first case is higher than the second, as is the default), maybe mixing tabs and spaces?

Reply:
Ah, yes, a fight with the text editor. Fixed.